EToCalculator.ts

```diff
@@ -0,0 +1,62 @@
+import { EToData } from "./types";
+
+// The implementation of this algorithm was guided by a step-by-step breakdown (http://edis.ifas.ufl.edu/pdffiles/ae/ae45900.pdf)
+/**
+ * Calculates the reference evapotranspiration using the Penman-Monteith (FAO-56) method (http://www.fao.org/3/X0490E/x0490e07.htm).
+ *
+ * @param etoData The data to calculate the ETo with.
+ * @param elevation The elevation above sea level of the watering site (in meters).
+ * @return The reference evapotranspiration (in millimeters per day).
+ */
+export function calculateETo( etoData: EToData, elevation: number ): number {
+	const avgTemp = ( etoData.maxTemp + etoData.minTemp ) / 2;
+
+	// Adjust the wind speed to a 2m height.
+	const windSpeed = etoData.windSpeed * 4.87 / Math.log( 67.8 * etoData.windSpeedMeasurementHeight - 5.42 );
```

Comment:
I think all Weather Providers should normalise `windspeed` to 2m and then this routine can use the value directly.

Reply:
I think it's better to normalize it here so we don't need to repeat the same code in each WeatherProvider.

Reply:
A helper function avoids the repetition of code. I guess either we need to standardise the definition of `windSpeed` or provide an additional field in the return, i.e. `windSpeedMeasuringHeight`. Guess I was just trying to reduce the number of fields. Fairly relaxed about this.

Reply:
Adding a helper function sounds like a good idea, I'll do that so we can remove the `windSpeedMeasuringHeight`. I think I'm also going to add a helper function to approximate radiation from cloud coverage so we won't need to repeat that code either.